src/inference/src/dev/core_impl.hpp

```diff
@@ -252,22 +254,34 @@ class CoreImpl : public ov::ICore, public std::enable_shared_from_this<ov::ICore
 
     std::shared_ptr<ov::Model> read_model(const std::string& model_path, const std::string& bin_path) const override;
 
-    ov::SoPtr<ov::ICompiledModel> compile_model(const std::shared_ptr<const ov::Model>& model,
-                                                const std::string& device_name,
-                                                const ov::AnyMap& config = {}) const override;
+    ov::SoPtr<ov::ICompiledModel> compile_model(
+        const std::shared_ptr<const ov::Model>& model,
+        const std::string& device_name,
+        const ov::AnyMap& config = {},
+        const std::function<std::string(const std::string&)>& encryption_func = {},
+        const std::function<std::string(const std::string&)>& decryption_func = {}) const override;
```

Comment:
why is it not a part of `ov::AnyMap config` ?

Reply:
Does it mean that it will be better to put encryption/decryption callback function as a new property?